### ISSUE TITLE
Multiple ApiExtraModels decorators

### ISSUE THREAD

**Issue by mo4islona** [nestjs/swagger#1366]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Consequent `ApiExtraModels` have singleton pattern. Last decorator on same method/controller overrides any previous. This causes use on this decorator ONLY on top-level. You cant create composed decorator which contains  `ApiExtraModels` inside   

```js

@Controller()
export class Controller {
  @ApiExtraModels(Model1)
  @ApiExtraModels(Model2)
  getList() {
      //...
  }
}
```
 Only `Model2` will be in a swagger docs.
 
 
 
## Expected behavior

Any `ApiExtraModels` decorator merges with all previous models  on method/controller

```js

@Controller()
export class Controller {
  @ApiExtraModels(Model1)
  @ApiExtraModels(Model2)
  getList() { 
    //...
   }
}
```
 Both `Model1` and `Model2` should be in a swagger docs.

## What is the motivation / use case for changing the behavior?

It is not possible to create composed decorators like below: 

```js
@Controller()
export class Controller {
@ApiSuccesslHttpResponse(Verification)
@ApiErrorHttpResponse(TooManyRequestsException, ThrottleVerification)
@ApiErrorHttpResponse(BadRequestException, VerificationInvalidCode)
 applyVerification() { 
    //...
   }
   
@ApiSuccesslHttpResponse(Verification)
@ApiErrorHttpResponse(BadRequestException, Verfication)
updateVerificationById() { 
    //...
   }
   
}

export const ApiErrorHttpResponse = <Model extends Type>(
  httpException: (new () => HttpException) | HttpException,
  model?: Model,
) => {
  const ex =
    typeof httpException === 'function' ? new httpException() : httpException;

  if (model) {
    return applyDecorators(
      ApiResponse({
        status: ex.getStatus(),
        schema: {
          properties: {
            error: {
              type: 'string',
              example: ex.getError(),
            },
            payload: {
              $ref: getSchemaPath(model),
            },
          },
        },
      }),
      ApiExtraModels(model),
    );
  }

  return ApiResponse({
    status: ex.getStatus(),
    schema: {
      properties: {
        error: {
          type: 'string',
          example: ex.getError(),
        },
      },
    },
  });
};
```

## Environment

<pre><code>
Nest version: 7.6.8

 
For Tooling issues:
- Node version: v14.17.0  
- Platform:   All

**Comment by micalevisk** [comment]
You can use `@ApiExtraModels(Model1, Model2)` tho

**Comment by mo4islona** [comment]
take look at motivation section please I explained why it is not convenient to have only such singleton pattern in real life

TLDR 
you cant use `@ApiExtraModels` in other decorators
only on root level as a standalone decorator

**Comment by kamilmysliwiec** [comment]
Let's track this here #1367 Thanks for reporting & creating a PR!
